README.md

```diff
@@ -1,583 +1,317 @@
+- `KO_DOCKER_REPO=my-dockerhub-user`
 
-The `--watch` flag (`-W` for short) does an initial `apply` as above, but as it
-does, it builds up a dependency graph of your program and starts to continuously
-monitor the filesystem for changes. When a file changes, it re-applies any yamls
-that are affected.
+# Build an Image
 
-For example, if I edit `github.com/foo/bar/pkg/baz/blah.go`, the tool sees that
-the `github.com/foo/bar/pkg/baz` package has changed, and perhaps both
-`github.com/foo/bar/cmd/one` and `github.com/foo/bar/cmd/two` consume that
-library and were referenced by `config/one-deploy.yaml` and
-`config/two-deploy.yaml`. The edit would effectively result in a re-application
-of:
+`ko publish ./cmd/app` builds and pushes a container image, and prints the resulting image digest to stdout.
 
 ```
-ko apply -f config/one-deploy.yaml -f config/two-deploy.yaml
+$ ko publish ./cmd/app
+...
+gcr.io/my-project/app-099ba5bcefdead87f92606265fb99ac0@sha256:6e398316742b7aa4a93161dce4a23bc5c545700b862b43347b941000b112ec3e
 ```
 
-This flag is still experimental, and feedback is very welcome.
-
-### `ko delete`
+Because the output of `ko publish` is an image reference, you can easily pass it to other tools that expect to take an image reference:
 
-`ko delete` simply passes through to `kubectl delete`. It is exposed purely out
-of convenience for cleaning up resources created through `ko apply`.
+To run the container:
+```
+docker run -p 8080:8080 $(ko publish ./cmd/app)
+```
 
-### `ko version`
+Or, for example, to deploy it to other services like [Cloud Run](https://cloud.google.com/run):
 
-`ko version` prints version of ko. For not released binaries it will print hash
-of latest commit in current git tree.
+```
+gcloud run deploy --image=$(ko publish ./cmd/app)
+```
 
-## With `minikube`
+## Configuration
 
-You can use `ko` with `minikube` via a Docker Registry, but this involves
-publishing images only to pull them back down to your machine again. To avoid
-this, `ko` exposes `--local` or `-L` options to instead publish the images to
-the local machine's Docker daemon.
+Aside from `KO_DOCKER_REPO`, you can configure `ko`'s behavior using a `.ko.yaml` file. The location of this file can be overridden with `KO_CONFIG_PATH`.
 
-This would look something like:
+### Overriding Base Images
 
-```shell
-# Use the minikube docker daemon.
-eval $(minikube docker-env)
+By default, `ko` bases images on `gcr.io/distroless/static:nonroot`. This is a small image that provides the bare necessities to run your Go binary.
 
-# Make sure minikube is the current kubectl context.
-kubectl config use-context minikube
+You can override this base image in two ways:
 
-# Deploy to minikube w/o registry.
-ko apply -L -f config/
+1. To override the base image for all images `ko` builds, add this line to your `.ko.yaml` file:
 
-# This is the same as above.
-KO_DOCKER_REPO=ko.local ko apply -f config/
+```yaml
+defaultBaseImage: my.registry/base/image
```

Comment:
Here (and everywhere) we should use `example.com` -- perhaps `registry.example.com`?

Reply:
Done.